src/components/PopupWithForm.js

```diff
@@ -5,12 +5,13 @@ export default class PopupWithForm extends Popup {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._popupInputs = this._popupForm.querySelectorAll(".modal__input");
+    this._saveButton = this._popupForm.querySelector(".modal__save-button");
     this._handleFormSubmit = handleFormSubmit;
   }
 
   close() {
     this._popupForm.reset();
-    this._popupForm.removeEventListener("submit", this._submitForm);
+    // this._popupForm.removeEventListener("submit", this._submitForm);
     super.close();
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`